modules/ROOT/pages/sap/sap-connector-reference-44.adoc

```diff
@@ -0,0 +1,807 @@
+| Gateway Service a| String |  Either the name or service port for the gateway service of the SAP system. |  | x
+| Program ID a| String |  SAP system program ID that is registered on the gateway. Must be unique for the given gateway. |  | x
+| Connection Count a| Number |  The number of connections to register on the gateway. |  1 | 
+| Targeted Function a| String |  Filters the incoming BAPI functions by their names. The name of the function you want to receive.  |  | 
+| Transactional Action a| Enumeration, one of:
+
+** ALWAYS_BEGIN
+** NONE |  The type of beginning action that sources can take regarding transactions |  NONE | 
+| Transaction Type a| Enumeration, one of:
+
+** LOCAL
+** XA |  The type of transaction to create. Availability depends on Mule version. |  LOCAL | 
+| Primary Node Only a| Boolean |  Whether this source should be executed only on the primary node when running in a cluster |  | 
+| Streaming Strategy a| * <<repeatable-in-memory-stream>>
+* <<repeatable-file-store-stream>>
+* non-repeatable-stream |  Configure to use repeatable streams. |  | 
+| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message |  | 
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  | 
+| Response a| Binary |  |  `#[payload]` | 
+|===
+
+==== Output
+[%autowidth.spread]
+|===
+|Type |Binary
+| Attributes Type a| <<SapAttributes>>
+|===
+
+=== For Configurations
+* <<inbound-config>> 
+
+
+
+== Types
+[[MessageServer]]
+=== Message Server
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Host a| String | The host of the message server |  | x
+| System Id a| String | System ID of the SAP system |  | x
+| Port a| Number | The port with which the connector logs into the message server |  | 
+| Group a| String | Group of SAP application servers |  | 
+| Router a| String | SAP router string to use for a system protected by a firewall |  | 
+|===
+
+[[Reconnection]]
+=== Reconnection
+
+The Reconnection Strategy settings are not enabled. Although disabled, reconnection is automatic when a computer running SAP Connector is disconnected and reconnected. Reconnection occurs based on default conditions established in the SAP JCo libraries. However, if the computer is accessing the SAP instance via VPN, reconnection fails and SAP Connector displays `TIMEOUT` and `CANNOT_SEND_IDOC` errors.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> | The reconnection strategy to use. |  | 
+|===
+
+[[reconnect]]
```

Comment:
Replace reconnect and reconnect forever with:
[[reconnect]]
=== Reconnect

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often to reconnect (in milliseconds). | |
| Count a| Number | The number of reconnection attempts to make. | |
| blocking |Boolean |If false, the reconnection strategy runs in a separate, non-blocking thread. |true |
|===

[[reconnect-forever]]
=== Reconnect Forever

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often in milliseconds to reconnect. | |
| blocking |Boolean |If false, the reconnection strategy runs in a separate, non-blocking thread. |true |
|===